config/src/providers/remappings.rs

```diff
@@ -6,12 +6,46 @@ use figment::{
 };
 use std::{
     borrow::Cow,
-    collections::{hash_map::Entry, HashMap},
+    collections::{btree_map::Entry, BTreeMap},
     fs,
     path::{Path, PathBuf},
 };
 use tracing::trace;
 
+/// Wrapper types over a `Vec<Remapping>` that only appends unique remappings.
+#[derive(Debug, Clone, Default)]
+pub struct Remappings {
+    /// Remappings.
+    pub remappings: Vec<Remapping>,
+}
+
+impl Remappings {
+    /// Create a new `Remappings` wrapper with an empty vector.
+    pub fn new() -> Self {
+        Self { remappings: Vec::new() }
+    }
+
+    /// Create a new `Remappings` wrapper with a vector of remappings.
+    pub fn new_with_remappings(remappings: Vec<Remapping>) -> Self {
+        Self { remappings }
+    }
+
+    /// Push an element ot the remappings vector, but only if it's not already present.
+    pub fn push(&mut self, remapping: Remapping) {
+        if self.remappings.iter().any(|existing| existing.name.contains(&remapping.name)) {
```

Comment:
I don't think is matches the previous dedup behaviour which was dedup by context and path?